src/AppInstallerCLICore/Workflows/DependenciesFlow.cpp

```diff
@@ -133,6 +135,104 @@ namespace AppInstaller::CLI::Workflow
         }
     }
 
+    void EnableWindowsFeaturesDependencies(Execution::Context& context)
+    {
+        if (!Settings::ExperimentalFeature::IsEnabled(Settings::ExperimentalFeature::Feature::WindowsFeature))
+        {
+            return;
+        }
+
+        const auto& rootDependencies = context.Get<Execution::Data::Installer>()->Dependencies;
+
+        if (rootDependencies.Empty())
+        {
+            return;
+        }
+
+        if (rootDependencies.HasAnyOf(DependencyType::WindowsFeature))
+        {
+            context << Workflow::EnsureRunningAsAdmin;
+            if (context.IsTerminated())
+            {
+                return;
+            }
+
+            HRESULT hr = S_OK;
+            DismHelper dismHelper = DismHelper();
+
+            bool force = context.Args.Contains(Execution::Args::Type::Force);
+            bool rebootRequired = false;
+
+            auto info = context.Reporter.Info();
+            auto warn = context.Reporter.Warn();
+
+            rootDependencies.ApplyToType(DependencyType::WindowsFeature, [&hr, &dismHelper, &info, &warn, &force, &rebootRequired](Dependency dependency)
+                {
+                    if (SUCCEEDED(hr) || force)
+                    {
+                        auto featureName = dependency.Id();
+                        DismHelper::WindowsFeature windowsFeature = dismHelper.CreateWindowsFeature(featureName);
+
+                        if (windowsFeature.DoesExist())
+                        {
+                            if (!windowsFeature.IsEnabled())

```

Comment:
I think there's a missing check for a pending reboot if the feature is already enabled?

Thinking of the case where a user attempts an install and gets a failure that a reboot is required. If they attempt to run the install command again without rebooting, the Windows Feature will be shown as enabled(?) and therefore won't trigger the rebootRequired, effectively negating the need to use `--force` by running the command twice.